typespec-extension/package.json

```diff
@@ -45,31 +45,31 @@
     "target/azure-typespec-extension-jar-with-dependencies.jar"
   ],
   "peerDependencies": {
-    "@typespec/compiler": ">=0.45.0 <1.0.0",
-    "@typespec/rest": ">=0.45.0 <1.0.0",
-    "@typespec/openapi": ">=0.45.0 <1.0.0",
-    "@typespec/versioning": ">=0.45.0 <1.0.0",
-    "@azure-tools/typespec-azure-core": ">=0.31.0 <1.0.0",
-    "@azure-tools/typespec-client-generator-core": ">=0.31.0 <1.0.0"
+    "@typespec/compiler": ">=0.49.0 <1.0.0",
+    "@typespec/rest": ">=0.49.0 <1.0.0",
+    "@typespec/openapi": ">=0.49.0 <1.0.0",
+    "@typespec/versioning": ">=0.49.0 <1.0.0",
+    "@azure-tools/typespec-azure-core": ">=0.35.0 <1.0.0",
+    "@azure-tools/typespec-client-generator-core": ">=0.35.0 <1.0.0"
```

Comment:
Required dependencies version bumped to 0.49, as emitter now uses the new `response.statusCodes`